src/superannotate/lib/core/config.py

```diff
@@ -0,0 +1,67 @@
+import threading
+from typing import Dict
+
+from dataclasses import dataclass
+from dataclasses import field
+
+
+class Session:
+    def __init__(self):
+        self.pk = threading.get_ident()
+        self._data_dict = {}
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, type, value, traceback):
+        if type is not None:
+            return False
+
+    def __del__(self):
+        Config().delete_current_session()
+
+    @property
+    def data(self):
+        return self._data_dict
+
+    @staticmethod
+    def get_current_session():
+        return Config().get_current_session()
+
+    def __setitem__(self, key, item):
+        self._data_dict[key] = item
+
+    def __getitem__(self, key):
+        return self._data_dict[key]
+
+    def __repr__(self):
+        return repr(self._data_dict)
+
+    def clear(self):
+        return self._data_dict.clear()
+
+
+class Singleton(type):
+    _instances = {}
+
+    def __call__(cls, *args, **kwargs):
```

Comment:
Is this singleton thread-safe? Shouldn't we have something like 
```
class Signleton(type):
    _lock = threading.Lock()
    _instances = {}
    def __call__(cls, *args, **kwargs):
         if cls not in cls._instances:
             with cls.lock:
                 if cls not in cls._instances:
                     cls._instances[cls] = super(Singleton, cls).__call__(*args, **kwargs)
         return cls._instances[cls]
```

Reply:
good point, thanks 
updating right now